src/iter.rs

```diff
@@ -0,0 +1,96 @@
+//! Additional `Bump`-related functionality for iterators.
+
+use crate::Bump;
+
+/// `Iterator` extensions for `Bump`-allocated types.
+pub trait BumpIterator: Iterator {
```

Comment:
Another idea that could go in this trait is `unzip_in(&Bump)`. `Iterator::unzip` requires the collections to implement `Default + Extend` -- maybe there could be a `DefaultIn` trait building from a bump allocator? This would also enable adaptations of `Iterator::partition` and `Itertools::partition_map`.